helix-term/src/ui/document.rs

```diff
@@ -54,10 +38,7 @@ impl<H: Iterator<Item = HighlightEvent>> Iterator for StyleIter<'_, H> {
                 HighlightEvent::HighlightEnd => {
                     self.active_highlights.pop();
                 }
-                HighlightEvent::Source { start, end } => {
-                    if start == end {
-                        continue;
-                    }
+                HighlightEvent::Source { end, .. } => {
```

Comment:
Why is it safe to ignore zero-width highlights now? (Or has it always been but we've just been skipping them?)

Reply:
Empty highlights are still being äskipped just differently. If you look at where this function is actually called it's now a while loop (that skips highlights as long as the end if <= current_pos).

This is a a bugfix that will be eventually relevant for conceal (this PR contained a prototype for that originally but kept all the bugfixes that were standalone commits). Where it's possible that part of th text will be skipped by the doc formatter. That while loop automatically handles empty highlights too so no need for the special case here (and in general it felt like a better solution)

Reply:
That is a8dbdd4df7a125a988a8ac0a0253a3ceab827a64


That and the rc => Vec switch should have probably been split of from this PR (back when I backported some fizes from this PR) but I probably forgot about those two since they are that relevant right now.